fv3gfs/util/utils.py

```diff
@@ -80,9 +80,9 @@ def safe_assign_array(
             raise
 
 
-def device_synchronize(array: Union[np.ndarray, Storage]):
+def device_synchronize(array: Optional[Union[np.ndarray, Storage]]):
```

Comment:
It makes sense to make this a default behavior. Could you extend the docstring to describe the conditional behaviors?

```suggestion
def device_synchronize(array: Optional[Union[np.ndarray, Storage]]=None):
```

Reply:
I entirely removed the array. The original idea was to mimic the streaming system of CUDA, but the actual solution will probably look vastly different. Therefore I removed it and made it clearly a full device sync.